### ISSUE TITLE
After upgrade to version 1.1.29 AMT devices are not anymore grouped with agent devices

### ISSUE THREAD

**Issue by vitko-bg** [Ylianst/MeshCentral#6386]
**Describe the bug**
Upgraded from 1.1.27 to 1.1.29 and now AMT devices are in group AMT only devices. Before upgrade device was showing as Agent + AMT on one row. Now same device with agent and AMT are in two gorups

**To Reproduce**
just upgrade from 1.1.27 to 1.1.29

**Expected behavior**
To have one device as Agent + AMT

**Screenshots**
![image](https://github.com/user-attachments/assets/fbbe37ff-833a-48f6-bf9c-103ccb4d94a4)

Would like to have "Agent + AMT" as before

**Server Software (please complete the following information):**
 - OS: Debian
 - Virtualization: NO
 - Network: WAN, reverse proxy - nginx

**Client Device (please complete the following information):**
 - Device: Desktop
 - OS: Windows 10 + AMT, Linux Debian + AMT
 - Network: Local to Meshcentral

**Comment by si458** [comment]
did you move agents around at all?
like move the groups around depending on where they was setup?
move device 1 from group1 to group2 but installed the meshagent in group1?

this can sometimes happen where the agents revert back to the orignal group they was installed to

**Comment by vitko-bg** [comment]
No, no agents moved. Straight after upgrade just logged on and chenges appeared instantly. AMT + Agent device spplit into two groups, the original one with agent and AMT only group appeared with the device as AMT only device

**Comment by si458** [comment]
thats bloody weird? we didnt change anything amt wise?
but im not 100% sure im afraid?
it might be that the AMT has de-cuppled itself from the meshid it had and gave itself a new one.
this can happen if the remote devices generated new meshids (say the web ssls changed or new network interfaces etc)

**Comment by vitko-bg** [comment]
Well, I deleted my group for AMT only devices to try and get again agent + amt device in one row. No luck.
So here is the screenshot from the device:
![image](https://github.com/user-attachments/assets/ec95c3d9-f145-4cd2-8864-9013822bdf58)

When I go to AMT tab, the result is below:

![image](https://github.com/user-attachments/assets/3133d28a-fb4b-4e20-b007-160df69113a6)

Any advise what else can I try to restore my AMT function ?

**Comment by si458** [comment]
sadly not. im in the same boat.
i removed the meshagent by mistake without thinking about it
then reinstalled it, now i have 1 devices, 1 for AMT and 1 for Agent :( 
we need like a merge feature to link the 2 devices together,
buts its very hard/complex to do something like this

the only thing i can suggest is install meshagent, deactivate AMT (factory reset amt if possible)
then try reactivating it using the meshagent and see if it can link to the new meshagent device

**Comment by vitko-bg** [comment]
ahaaaa, so it is a global issue happened :) Interesting that 1.1.27 was fine. Will try your suggest, hope in next couple of days as have to reboot the device I have for AMT reset and will update you.

**Comment by vitko-bg** [comment]
@si458 - still the same. Tried to reset AMT, remove agent , re-add agent and AMT connections both ways, first AMT and after agent, and vice versa, still same. It is all in the latest version 1.1.43 remain as the issue.
Not sure if there are plans to fix this and to have agent and AMT connection in one line, but if there are no plans this to be fixed so there is no point to have "Intel AMT" tab on the device which is agent connected, it just cannot be used.

**Comment by si458** [comment]
@vitko-bg im still having this problem every so often with my 2 devices too, i havent found a fix yet sadly 👎 
the one device with CIRA on it seems to decide to seperate and go into another device group and i just dont know why

**Comment by vitko-bg** [comment]
Any update on this please? Is anyone else having the same issue and if managed to fix please provide some fox for this please.

**Comment by vitko-bg** [comment]
Hi @si458 , it is now back!!!! 

<img width="108" height="146" alt="Image" src="https://github.com/user-attachments/assets/6b8e21eb-8c33-45cf-abd2-e1e125a60719" />

I just updated from 1.1.46 to 1.1.48 and issue has been resolved 👍

**Comment by si458** [comment]
we havent changed anything code wise for the groups haha
i think the issue is because the AMT cant tell meshcentral what its ID is to do linking etc
so it can from time to time get mixed up
but see how you get on
if it doesnt return after say 48 hours then please can you close the issue 👍 
we can always re-open it months later if it starts again haha

**Comment by vitko-bg** [comment]
Yes, for sure will close it if all is OK. BTW, you mentioned that on your side is the same, what about after update to 1.1.48? If your server can see Agent + CIRA, then it must be the update, if not - well ...... then possibly issue is still persist.

**Comment by si458** [comment]
To be honest I haven't noticed in donkeys, I only have a single machine with amt+agent setup and it's always said amt+agent for months now
